### PR TITLE
Adding balena SDK and routes

### PR DESCRIPTION
**Pull request by nucleardreamer**
This PR adds some basic device info routes + ui via the balena SDK, and adds routes for get/set/deleting environment variables on the device as well, to tee up UI being implemented.

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -45,3 +43,5 @@ COPY --from=expressjs-build-step /build-context .
 COPY expressjs/src src
 
 CMD ["node", "src/index.js"]
+
+ENV NODE_ENV=production
```
Comment (maggie0002) [inline Dockerfile.template:47]: Is there a particular reason this is down at the bottom? It usually features at the top, like the build step before.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,15 +1,16 @@
 version: '2.1'
 
 services:
-  balena-cli-ui:
+  backend:
```
Comment (maggie0002) [inline docker-compose.yml:4]: This container also has the Quasar UI in it. Calling it backend might be a bit confusing?

File: ui/src/components/DeviceInfo.vue [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          <span>{{ (response.data.storage_usage / 1000).toFixed(2) }}GB / {{ (response.data.storage_total / 1000).toFixed(2) }}GB</span>
+          <q-linear-progress color="secondary" rounded :value="response.data.storage_usage / response.data.storage_total" class="q-mt-md" />
+        </q-card-section>
+      </q-card>
+    </div>
+    <hr class="q-mt-xl q-mb-lg" />
+    <q-expansion-item
+      expand-separator
+      icon="code"
+      label="Response Details"
```
Comment (maggie0002) [inline ui/src/components/DeviceInfo.vue:40]: Could this above and below be an i18n entry for future translation?

File: ui/src/layouts/MainLayout.vue [hunk 4]
```diff
@@ -74,8 +75,8 @@ export default defineComponent({
 
     const menuList = [
       {
-        icon: 'home',
-        label: t('Home'),
+        icon: 'visibility',
+        label: t('Device Info'),
```
Comment (maggie0002) [inline ui/src/layouts/MainLayout.vue:79]: Doesn't look like this is in the i18n file.

**Comment by maggie0002** [comment]
Thanks for these, really exciting to see a whole new take on what this can do. It seems the possibilities are endless. 

I'm not familiar with the SDK, does it reach out to the Balena online servers to retrieve the info? I'm wondering what happens if someone accesses the interface while the device is offline, is there error handling for that?

File: docker-compose.yml (hunk 2 above)
Comment (nucleardreamer) [reply docker-compose.yml:4]: good thinking, I wasn't sure what to call it. maybe just `app` but that seems generic. Perhaps just `server` is explicit enough

**Comment by nucleardreamer** [comment]
Yes, the SDK is essentially a node wrapper library that make calls to the Balena API which uses [pine.js](https://github.com/balena-io/pinejs) under the hood. Currently there is no way to get all the device info from the supervisor, the cloud holds it as the source of truth and resolves all that info back down to the supervisor. The next step after this PR is to add offline states to the UI. It's not an ideal situation, but it's what we have to work with today!

**Comment by nucleardreamer** [comment]
Note, I also added `postman_collection.json` for an easy way to test out these endpoints while developing

**Comment by maggie0002** [comment]
Couple of other changes pushed, it is worth a quick glance at the commits to see the changes.

**Pull request merged**